scripts/train_lm.py

```diff
@@ -146,12 +146,39 @@ def WriteMetaparameters(metaparameters, ngram_order, num_train_sets, out_file):
 
 def FormatMetaparameters(metaparameters):
     out = []
+    # get the maximum length of decimals of metaparameters
+    decimal_length = 0
     for param in metaparameters:
-        x = '{:.3f}'.format(param)
-        if x == '0.00':
-            x = '{:.3g}'.format(param)
-        out.append(x)
-
+        temp_length = len(str(param))
+        if decimal_length < temp_length:
+            decimal_length = temp_length
+    decimal_length -= 2
+    
+    temp_original = metaparameters[0]
+    temp = round(temp_original, 3)
+    for param in metaparameters[1:]:
+        round_index = 3
+        x_original = param
+        x = round(x_original, round_index)
+        while(temp == x and round_index <= decimal_length):
```

Comment:
and please add comments so it's clear what's happening and why.

On Tue, Aug 16, 2016 at 9:23 PM, chris920820 <notifications@github.com>
wrote:

> In scripts/train_lm.py
> <https://github.com/danpovey/pocolm/pull/52#discussion_r75060772>:
>
> > -        if x == '0.00':
> > -            x = '{:.3g}'.format(param)
> > -        out.append(x)
> > -
> > +        temp_length = len(str(param))
> > +        if decimal_length < temp_length:
> > +            decimal_length = temp_length
> > +    decimal_length -= 2
> > +
> > +    temp_original = metaparameters[0]
> > +    temp = round(temp_original, 3)
> > +    for param in metaparameters[1:]:
> > +        round_index = 3
> > +        x_original = param
> > +        x = round(x_original, round_index)
> > +        while(temp == x and round_index <= decimal_length):
>
> Sounds reasonable. I will keep trying a right but elegant way if possible.
> best regards, Zhouyang
> … <#m_2371658173570960292_>
> On Aug 17, 2016, at 12:08 AM, Wang Jian ***@***.***> wrote: In
> scripts/train_lm.py: > - if x == '0.00': > - x = '{:.3g}'.format(param) > -
> out.append(x) > - > + temp_length = len(str(param)) > + if decimal_length <
> temp_length: > + decimal_length = temp_length > + decimal_length -= 2 > + >
> + temp_original = metaparameters[0] > + temp = round(temp_original, 3) > +
> for param in metaparameters[1:]: > + round_index = 3 > + x_original = param
> > + x = round(x_original, round_index) > + while(temp == x and round_index
> <= decimal_length): I agree with Dan, generally we should avoid any tricky
> things, unless it can improve performance very much. — You are receiving
> this because you commented. Reply to this email directly, view it on
> GitHub, or mute the thread.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/danpovey/pocolm/pull/52/files/3bb42c2f1362ffd85d6a4cf0b1fe1a50ab2fe9e1#r75060772>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu7c3kYWjePNOcf4MpUJuZPI1HR9Gks5qgoysgaJpZM4Jl6Bt>
> .
>